Code/Tools/ProjectManager/Source/ProjectExportController.cpp

```diff
@@ -0,0 +1,146 @@
+            if (logFilePathQuery.IsSuccess())
+            {
+                projectButton->SetBuildLogsLink(QUrl::fromLocalFile(logFilePathQuery.GetValue()));
+            }
+            projectButton->SetState(ProjectButtonState::Exporting);
+
+            if (!m_lastLine.isEmpty())
+            {
+                UpdateUIProgress(m_lastLine);
+            }
+        }
+    }
+
+    const ProjectInfo& ProjectExportController::GetProjectInfo() const
+    {
+        return m_projectInfo;
+    }
+
+    void ProjectExportController::UpdateUIProgress(const QString& lastLine)
+    {
+        m_lastLine = lastLine.left(s_maxDisplayedBuiltOutputChars);
+
+        if (m_projectButton)
+        {
+            m_projectButton->SetContextualText(m_lastLine);
+        }
+    }
+
+    void ProjectExportController::HandleResults(const QString& result)
+    {
+        if (!result.isEmpty())
+        {
+            AZ::Outcome<QString, QString> logFilePathQuery = m_worker->GetLogFilePath();
+            if (result.contains(tr("log")))
+            {
+                QMessageBox::StandardButton openLog = QMessageBox::critical(
+                    m_parent,
+                    tr(LauncherExportFailedMessage),
+                    result + tr("\n\nWould you like to view log?"),
+                    QMessageBox::No | QMessageBox::Yes);
+
+                if (openLog == QMessageBox::Yes)
+                {
+                    // Open application assigned to this file type
+                    if (!logFilePathQuery.IsSuccess() || !QDesktopServices::openUrl(QUrl::fromLocalFile(logFilePathQuery.GetValue())))
+                    {
+                        qDebug() << "QDesktopServices::openUrl failed to open " << m_projectInfo.m_logUrl.toString() << "\n";
+                    }
+                }
+                
+                if (logFilePathQuery.IsSuccess())
+                {
+                    m_projectInfo.m_logUrl = QUrl::fromLocalFile(logFilePathQuery.GetValue());
+                }
+            }
+            else
+            {
+                if (logFilePathQuery.IsSuccess())
+                {
+                    QMessageBox::critical(m_parent,
+                                      tr("%1\nYou can check the logs in the following directory:\n%2")
+                                        .arg(LauncherExportFailedMessage)
+                                        .arg(logFilePathQuery.GetValue()),
+                                      result);
+                }
+                else
+                {
+                    QMessageBox::critical(m_parent,
+                                      tr("%1\nNo logs are available at this time.")
+                                        .arg(LauncherExportFailedMessage),
+                                      result);
+                }
+            }
+
+            emit Done(false);
+            return;
+        }
+
+        emit Done(true);
```

Comment:
nit: you could just create a boolean variable `success` and default it to true at the beginning of this function, and replace `emit Done(false)` with `success = false;`, so that at the end of this function you could just do `emit Done(success);` and remove the line
```
  emit Done(false);
  return;
```